sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/link.py

```diff
@@ -232,8 +232,15 @@ def _incoming_detach(self, frame) -> None:
         # TODO: on_detach_hook
         if frame[2]:  # error
             # frame[2][0] is condition, frame[2][1] is description, frame[2][2] is info
-            error_cls = AMQPLinkRedirect if frame[2][0] == ErrorCondition.LinkRedirect else AMQPLinkError
-            self._error = error_cls(condition=frame[2][0], description=frame[2][1], info=frame[2][2])
+            condition = frame[2][0]
+            error_cls = AMQPLinkRedirect if condition == ErrorCondition.LinkRedirect else AMQPLinkError
+
+            # description and info are optional fields, from the AMQP spec.
+            # https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#type-error
+            description = None if len(frame[2]) < 2 else frame[2][1]
```

Comment:
Is using length the right way to detect whether these fields are set? If description is missing but info is set, is description omitted from the frame entirely? If so, in this case, the length is 3. This would mean that info in the frame is getting set to description, based on the length.

Will take a look at the actual frame that's coming over to check.

Reply:
You're right, I wasn't sure how the list was filled out and should have looked.

Ideally, the list should always be 3 elements, and the individual slots should be None if they're not there. Or maybe just a tuple or something of that nature? I didn't trace to see where it was coming from.